gpkit/substitution.py

```diff
@@ -191,23 +189,17 @@ def substitution(nomial, substitutions, val=None):
                                          " those of the original '%s' [%s]." %
                                          (sub, sub.units.units,
                                           var, var.units.units))
-                exps_[i] += HashVector({sub: x})
-                varlocs_[sub].append(i)
-            elif isinstance(sub, Monomial):
-                if isinstance(var.units, Quantity):
-                    try:
-                        new_units = var.units/sub.units
-                        cs_[i] *= new_units.to('dimensionless')
-                    except DimensionalityError:
-                        raise ValueError("units of the substituted monomial"
-                                         " '%s' [%s] are not compatible with"
-                                         " those of the original '%s' [%s]." %
-                                         (sub, sub.units.units,
```

Comment:
note that for monomials this will print things like

```
units of the substituted monomial x [m] [m] are not compatible with the original y [volt]
```

Because by default monomials print with units and varkeys do not. 

To fix this, replace this line with `(sub.str_without(["units"]), sub.units.units,`